lib/marc_cleanup/variable_fields.rb

```diff
@@ -354,40 +354,54 @@ def x00_subfd_no_comma?(record)
   end
 
   def relator_comma?(record)
-    comma_regex = /^.*[^,]$/
-    record.fields.each do |field|
-      next unless field.tag =~ /[17][01][01]/
-
-      code_array = ''
-      field.subfields.each do |subfield|
-        code_array << subfield.code
-      end
-      if field.tag =~ /[17][01]0/
-        subfx_e_index = code_array.index(/.e/)
-        return true if subfx_e_index && field.subfields[subfx_e_index].value =~ comma_regex
-      elsif field.tag =~ /[17]11/
-        subfx_j_index = code_array.index(/.j/)
-        return true if subfx_j_index && field.subfields[subfx_j_index].value =~ comma_regex
-      end
+    record.fields(%w[100 110 111 700 710 711]).each do |field|
+      relator_index = relator_subfield_index(field)
+      next unless relator_index
+      return true if field.subfields[relator_index - 1].value =~ /[^,]$/
     end
     false
```

Comment:
I believe this could also be refactored to use `#any?`, rather than `#each?`, which from my perspective makes the goal of the block a little clearer:

```suggestion
    record.fields(%w[100 110 111 700 710 711]).any? do |field|
      relator_index = relator_subfield_index(field)
      next unless relator_index
      field.subfields[relator_index - 1].value =~ /[^,]$/
    end
```